shared/js/background/declarative-net-request.js

```diff
@@ -567,6 +573,44 @@ export function flushSessionRules () {
     })
 }
 
+/**
+ * Find the session rule with the given ID, if it exists.
+ * @param {number} id Rule ID
+ * @returns {Promise<chrome.declarativeNetRequest.Rule | undefined>}
+ */
+async function findSessionRuleFromId (id) {
```

Comment:
I guess we still need this for now, but fyi this was just fixed https://bugs.chromium.org/p/chromium/issues/detail?id=1379699

Reply:
Yea, also see `findExistingDynamicRule` above which does nearly the same. Perhaps you could expand the scope of `findExistingDynamicRule` to handle both session and dynamic rules?